File: DarkOrbit 10.0/Net/netty/commands/AudioSettingsModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class AudioSettingsModule
    {
        public const short ID = 27262;

        public bool NotSet = false;
        public bool PlayCombatMusic = false;
        public int Voice = 0;
        public int Sound = 0;
        public int Music = 0;

        public AudioSettingsModule(bool pNotSet, int pSound, int pMusic, int pVoice, bool pPlayCombatMusic)
        {
            this.NotSet = pNotSet;
            this.Sound = pSound;
            this.Music = pMusic;
            this.Voice = pVoice;
            this.PlayCombatMusic = pPlayCombatMusic;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(-4407);
            param1.writeInt(this.Voice << 9 | this.Voice >> 23);
            param1.writeInt(this.Sound << 7 | this.Sound >> 25);
            param1.writeBoolean(this.PlayCombatMusic);
            param1.writeInt(this.Music >> 4 | this.Music << 28);
            param1.writeShort(-5063);
            param1.writeBoolean(this.NotSet);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/AudioSettingsModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

public class AudioSettingsModule
        implements ServerCommand {

    public static int     ID               = 6686;
    public        boolean mNotSet          = false;
    public        boolean mPlayCombatMusic = false;
    public        int     mVoice           = 0;
    public        int     mSound           = 0;
    public        int     mMusic           = 0;

    public AudioSettingsModule(boolean pNotSet, int pSound, int pMusic, int pVoice, boolean pPlayCombatMusic) {
        this.mNotSet = pNotSet;
        this.mSound = pSound;
        this.mMusic = pMusic;
        this.mVoice = pVoice;
        this.mPlayCombatMusic = pPlayCombatMusic;
    }

    public int getID() {
        return ID;
    }

    public int method_1005() {
        return 14;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeInt(this.mSound << 16 | this.mSound >>> 16);
            param1.writeBoolean(this.mPlayCombatMusic);
            param1.writeInt(this.mVoice >>> 7 | this.mVoice << 25);
            param1.writeBoolean(this.mNotSet);
            param1.writeInt(this.mMusic >>> 8 | this.mMusic << 24);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}